src/backend/gl/src/device.rs

```diff
@@ -1548,8 +1595,23 @@ impl d::Device<B> for Device {
         }
     }
 
-    unsafe fn get_fence_status(&self, _: &n::Fence) -> Result<bool, d::DeviceLost> {
-        unimplemented!()
+    unsafe fn get_fence_status(&self, fence: &n::Fence) -> Result<bool, d::DeviceLost> {
+        let gl = &self.share.context;
+
+        let mut len = 0;
+        let mut values = [0];
+        gl.GetSynciv(fence.0.get(), gl::SYNC_STATUS, values.len() as i32, &mut len, values.as_mut_ptr());
+
+        if len == 0 {
+            if let Err(err) = self.share.check() {
+                error!("Error getting fence status: {:?}", err);
```

Comment:
nit: could just be `error!("Error getting fence status: {:?}", self.share.check());`